src/coreclr/vm/callstubgenerator.cpp

```diff
@@ -1453,6 +1457,53 @@ void CallStubGenerator::ComputeCallStub(MetaSig &sig, PCODE *pRoutines)
         ArgLocDesc argLocDesc;
         argIt.GetArgLoc(ofs, &argLocDesc);
 
+        // Each argument takes at least one slot on the interpreter stack
+        int interpStackSlotSize = INTERP_STACK_SLOT_SIZE;
+
+        // Each entry on the interpreter stack is always aligned to at least 8 bytes, but some arguments are 16 byte aligned
+        TypeHandle thArgTypeHandle;
+        if (argIt.GetArgType(&thArgTypeHandle) == ELEMENT_TYPE_VALUETYPE)
+        {
+            unsigned align = CEEInfo::getClassAlignmentRequirementStatic(thArgTypeHandle);
+            if (align < INTERP_STACK_SLOT_SIZE)
+            {
+                align = INTERP_STACK_SLOT_SIZE;
+            }
+            else if (align > INTERP_STACK_ALIGNMENT)
+            {
+                align = INTERP_STACK_ALIGNMENT;
+            }
+            assert(align == 8 || align == 16); // At the moment, we can only have an 8 or 16 byte alignment requirement here
+            while (interpreterStackOffset != ALIGN_UP(interpreterStackOffset, align))
```

Comment:
A nit - it seems "if" instead of while would be sufficient. If the align is 8, there is nothing to do and if the align is 16, there may be a need to inject one alignment. 
If this is a preparation for possible larger alignment support in the future, then when that happens, maybe we could parametrize the InjectInterpStackAlign by the size to add to the interpreter stack pointer instead of injecting multiple alignments.